### PR TITLE
Add revup topics bewteen HEAD and relative branch to revup commit string comment

### PR DESCRIPTION
**Pull request by malcolm-leclair-skydio**
The output will look like this

```
user@host:~/revup$ revup amend --insert
\# Please enter the commit message for your changes. Lines starting
\# with '#' will be ignored, and an empty message aborts the commit.
\#
\# Topics found between HEAD and origin/master
\#  my_topic
\#  my_other_topic
\#
```

Topic: revup_topic_in_commit_message

### CODE REVIEW

**Comment by malcolm-leclair-skydio** [comment]
Reviews in this chain:
└**https://github.com/Skydio/revup/pull/53** **Add revup topics to revup commit string using commits from base/relative branch**

**Comment by malcolm-leclair-skydio** [comment]
| # | head | base | diff | date | summary |
| - | - | - | - | - | - |
| 0 | [bf3a61d4](/Skydio/revup/commit/bf3a61d40e46e95489c6c269472835586040aee3) | [1ec4bcaa](/Skydio/revup/commit/1ec4bcaa6e1223048a02c0bf0f578655fee72f3b) | [diff](/Skydio/revup/compare/1ec4bcaa6e1223048a02c0bf0f578655fee72f3b..bf3a61d40e46e95489c6c269472835586040aee3) | Sep 20 12:46 PM |  2 files changed, 25 insertions(+), 2 deletions(-) |

File: revup/amend.py [hunk 1]
```diff
@@ -46,6 +46,27 @@ def invoke_editor_for_commit_msg(
...
+    topics = topic_stack.TopicStack(
+        git_ctx,
+        args.base_branch,
+        args.relative_branch,
+        None,
+        None,
+    )
+
+    try:
+        await topics.populate_topics()
```
Comment (jerry-skydio) [inline revup/amend.py:59]: populate_topics takes a decent bit of time, can we

- cache the result of populate_topics from parse_ref_or_topic so it only gets called once
- only get summary if args.parse_topics is true

File: revup/amend.py [hunk 2]
```diff
@@ -46,6 +46,27 @@ def invoke_editor_for_commit_msg(
...
+        topic_lines = "\n " + "\n ".join(topics.topics.keys()) + "\n"
+        return (
+            f"Current Topics from [base:{topics.base_branch}]"
+            f"[relative:{topics.relative_branch}]{topic_lines}"
+        )
+    except RevupUsageException:
+        # An error occured or no topic tags were found. Continue without topics
+        return "\nNo Topics Found\n"
```
Comment (jerry-skydio) [inline revup/amend.py:67]: just return "" here

File: docs/revup.md [hunk 3]
```diff
@@ -96,6 +96,10 @@ Revup is comprised of several sub-commands.
 : Modify a commit in the current stack using the contents of the
 cache. Can also change the commit text.
 
+**revup commit**
+: Create a commit in the current stack using the contents of the
+cache. Can also change the commit text. A convenience wrapper for revup amend --insert
```
Comment (jerry-skydio) [inline docs/revup.md:101]: Not necessary to duplicate this description, just say "A convenience wrapper for revup amend --insert"

**Comment by jerry-skydio** [comment]
same, also squash all commits before submitting

File: revup/amend.py (hunk 1 above)
Comment (malcolm-leclair-skydio) [reply revup/amend.py]: Both of these requests are added. While I was testing pulling up the editor with the commit was not noticeably slow for this repo, or our internal repo.

Comment (jerry-skydio) [reply revup/amend.py]: yeah i only mean slow in a relative sense. it takes about .2s which is significant in the sense that we want the tool to be as fast as reasonably possible

**Review by jerry-skydio — APPROVED** [review]

**Pull request merged**